packages/sanity/src/core/studio/telemetry/useMeasurePerformanceTelemetry.ts

```diff
@@ -0,0 +1,65 @@
+import {useTelemetry} from '@sanity/telemetry/react'
+import {useCallback, useEffect} from 'react'
+
+import {PerformanceINPMeasured} from '../__telemetry__/performance.telemetry'
+
+function getElementIdentifier(node: Node | null) {
+  if (!node) {
+    return null
+  }
+  if (!(node instanceof Element)) {
+    return node.nodeName
+  }
+  // Note: Deliberately using classList instead of className here since className isn't always a string
+  // See https://developer.mozilla.org/en-US/docs/Web/API/Element/className#notes
+  const {nodeName, classList, id} = node
+  return (
+    nodeName.toLowerCase() +
+    (id ? `#${id}` : '') +
+    (classList ? `.${classList.value.replaceAll(' ', '.')}` : '')
```

Comment:
I think I'd collect these two data attributes as well:
- `data-testid`
- `data-ui`

As the `classList` coming from `styled-components` can be botched/unusable if code minification is enabled (it is by default in `sanity build` and `next build`). And `id` sometimes might be using `useId()` or some other source that is difficult to trace back.

The `data-testid` is super high value since it's already used for E2E and gives useful context as to where and what the component is.
Same is true for `data-ui` as it's used by `@sanity/ui`, and we already know that primitives like `<Tooltip>`, `<Popover>` and `<Input>` are already hot paths.

Reply:
Good point. This was inspired by how it's reported in the Performance devtools panel, but agree data-test-id and data-ui is valuable. I do wonder if we want id and classlist at all, but feels like _some_ hints is better than none? I'll add dataUi and testId as additional props.

<img width="480" alt="image" src="https://github.com/user-attachments/assets/fbf86efe-428b-4b60-b0e9-88a391b6210e" />